code/modules/projectiles/projectile.dm

```diff
@@ -60,9 +62,15 @@
 	return ..()
 
 /obj/item/projectile/proc/Range()
-	range--
-	if(range <= 0 && loc)
-		on_range()
+    range--
+    if(damage > 0)

```

Comment:
I would do `if(damage > 0 && tile_dropoff)` since we don't need to calculate a substraction if there is specified dropoff value. Same with the other check.

Reply:
Good point, quickly tested it out to make sure everything still worked and I'm now committing it.